### ISSUE TITLE
Station search: Use hostname

### ISSUE THREAD

**Issue by CacioGit** [Suplanus/Cuterdio#107]
This is kind of a mix between a bug report and an enhancement request:
on both my Mac and my iPhone, when I start Cuterdio, I am greeted by a black screen, (an empty radio station list I imagine), and with a field where I can enter an URL to specify a new, custom radio station.

I can only assume that this may be due to me having tested early betas/releases of the app, and there may be some "orphaned" configuration files in my iCloud preferences for Cuterdio..

Is there a way to force a complete reset of the app? Simply uninstalling and re-installing it doesn't seem to solve my issue.

TIA

This is how my "start screen" looks like:

![Screenshot of ScreenFloat (18-08-20, 09-11-29)](https://user-images.githubusercontent.com/15092052/90481952-dec09980-e132-11ea-89d4-d6e71bef394b.jpg)

**Comment by Suplanus** [comment]
Every device has its own configuration (there is no sync yet).
So you have to add a station on every device.
Enter a search term in the search field and add the station.

Is that solving your problem?

**Comment by CacioGit** [comment]
Thank you for the quick answer, but I do not think this is my issue..

When I search for any radio station, nothing is ever returned:
![image](https://user-images.githubusercontent.com/15092052/90514023-e0a15180-e160-11ea-8439-45023f0d3517.jpeg)

Only if I add a custom defined station, I then see it listed:

![image](https://user-images.githubusercontent.com/15092052/90514219-2b22ce00-e161-11ea-80c7-a60103980b88.jpeg)

Is this how it is supposed to work?

I am seeing the same effect on both Mac and iPhone, so no pre-defined stations, and no easy way to pick one from the Search field.

**Comment by Suplanus** [comment]
I think you have to search for `SWR` and not `SWR3`:
![image](https://user-images.githubusercontent.com/2002916/90514755-e3e90d00-e161-11ea-9c8d-2a9bf0a20d13.png)

**Comment by CacioGit** [comment]
Mystery solved!
I noticed the following entries in the daily report from my FritzBox:
----

18.08.20 14:59:33 | Der Zugriff auf die feste IP-Adresse 45.77.62.161 wurde durch die Filter für Internetseiten verhindert.
-- | --
18.08.20 14:59:33 | Der Zugriff auf die feste IP-Adresse 95.179.139.106 wurde durch die Filter für Internetseiten verhindert.
18.08.20 14:58:12 | Der Zugriff auf die feste IP-Adresse 91.132.145.114 wurde durch die Filter für Internetseiten verhindert. [18 Meldungen seit 18.08.20 14:40:11]
----

Those IP addresses seems to be connected to: 
91.132.145.114, de1.api.radio-browser.info

I tested to skip the IP-address filtering of the FritzBox my using a VPN, and voila, now my Cuterio searches look like yours!

Bottom line:
you seem to be using hard-coded IP addresses to reach out to the RadioBrowser's API, and those requests are being filtered by a FritzBox working with filtering lists.

At least now I know the trigger of the problem.. I will leave it to you to decide if you can move away from hard-coded IP addresses or not

Thank you for looking into this anyway, really appreciated!

**Comment by Suplanus** [comment]
Oh wow, thanks 💕

The IP cames from DNS of radio-browser.info.  
There is a problem in the SDK I use so I have to had use IP and not the hostname / url.

Is the IP in your filter list or in a black list from some service?

**Comment by CacioGit** [comment]
I think this applies to everyone using a FritzBox as an home router:
as soon as you enable any kind of filter list (I use one for example for my kids internet access), fixed IP addresses are blocked by default, all of them, not just "dangerous" ones, so that all traffic has to go over DNS names (and can be filtered..).

It's in German, but this is the description of the issue from the vendor:
https://avm.de/service/fritzbox/fritzbox-7590/wissensdatenbank/publication/show/3400_FRITZ-Box-meldet-Der-Zugriff-auf-die-feste-IP-Adresse-wurde-durch-die-Filter-fur-Internetseiten-verhindert/

In my case I could put the radio-browser.info IP address in the "allowed" list.. or simply enable a VPN for Cuterio when I want to add a new station :)

Could be worth adding this to a FAQ somewhere, while you use the SDK's version with this "limitation".

You can mark the question as Resolved from my side.

**Comment by Suplanus** [comment]
Thanks again.

I try to find a workaround, so there is no need for an FAQ 🦄
